flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SplitAssignmentTracker.java

```diff
@@ -83,10 +86,8 @@ public void snapshotState(
      */
     public void restoreState(SimpleVersionedSerializer<SplitT> splitSerializer, DataInputStream in)
             throws Exception {
```

Comment:
It is also a good to read out and drop possible empty state. This will protect any future trailing state reader. But I guess in that stage, there will be a version bump to `SourceCoordinatorSerdeUtils.CURRENT_VERSION`. So I am neutral to this, just a preference to match write side.

Reply:
Thanks @kezhuw! I updated my code to read the map back in ```restoreStage()``` in 87b0f63